src/tribol/mesh/MfemData.hpp

```diff
@@ -345,8 +346,9 @@ class ParentField {
    * @brief Set a new transfer object when the redecomp mesh has been updated
    *
    * @param xfer Updated parent mesh to redecomp mesh transfer object
+   * @param use_device If true, use device memory for the redecomp grid function
```

Comment:
Would it ever be the case that you are running contact on the GPU, but NOT actually want to use device memory here? That is, do we need this boolean? Or, can this routine perform some execution policy check and then use device memory as required?